### PR TITLE
Add support for solaris

### PR DESCRIPTION
**Pull request by twiforce**

### CODE REVIEW

File: deps/libpcre/solaris/config.h [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+/* #undef SUPPORT_PCREGREP_JIT */
+
+/* Define to any value to enable support for Unicode properties. */
+/* #undef SUPPORT_UCP */
+
+/* Define to any value to enable support for the UTF-8/16/32 Unicode encoding.
+   This will work even in an EBCDIC environment, but it is incompatible with
+   the EBCDIC macro. That is, PCRE can support *either* EBCDIC code *or*
+   ASCII/UTF-8/16/32, but not both at once. */
+/* #undef SUPPORT_UTF */
```
Comment (calzoneman) [inline deps/libpcre/solaris/config.h]: cytubefilters creates and executes all regexes in UTF-8 mode, so if `SUPPORT_UTF` is undef'd (as it is here), then the module will compile but won't filter anything.  Can you try manually changing this to `#define SUPPORT_UTF` (like in https://github.com/calzoneman/cytubefilters/commit/7663b3a91a77f2ee68a1aea26a4c6ecf36817c02) and see if that works?

Comment (calzoneman) [inline deps/libpcre/solaris/config.h]: I should note that by "see if that works" I mean make the change, run `node-gyp configure` and `node-gyp build`, then run `npm test` and ensure that all tests pass.

Comment (twiforce) [inline deps/libpcre/solaris/config.h]: Yes, thank you, I just noticed that chatfilters were not working. Now everything is fine, at least as I can see.

**Comment by calzoneman** [comment]
Ok, I'm going to take your word for it as I don't have a solaris machine handy to test.

**Pull request merged**